src/main.ts

```diff
@@ -39,12 +39,22 @@ if (window.location.origin === 'https://hub.nimiq-testnet.com') {
   });
 }
 
-new Vue({
+const app = new Vue({
+  data: { loading: false },
   router,
   store,
   render: (h) => h(App),
 }).$mount('#app');
 
+router.beforeEach((to, from, next) => {
+  app.loading = true;
```

Comment:
On a fast internet connecting, this causes flickering.
I'd only set this after a short delay.